modules/apps/foundation/portal-configuration/portal-configuration-module-configuration/src/main/java/com/liferay/portal/configuration/module/configuration/internal/ConfigurationInvocationHandler.java

```diff
@@ -65,61 +72,136 @@ public Object invoke(Object proxy, Method method, Object[] args)
 		}
 		else if (returnType.equals(float.class)) {
-			return _typedSettings.getFloatValue(method.getName());
+			return _typedSettings.getFloatValue(key);
 		}
 		else if (returnType.equals(int.class)) {
-			return _typedSettings.getIntegerValue(method.getName());
+			return _typedSettings.getIntegerValue(key);
 		}
 		else if (returnType.equals(LocalizedValuesMap.class)) {
-			return _typedSettings.getLocalizedValuesMap(method.getName());
+			LocalizedValuesMap value = _typedSettings.getLocalizedValuesMap(
+				key);
+
+			if (getReturnTypeDefault) {
+				return value;
+			}
+
+			Map<Locale, String> valuesMap = value.getValues();
+
+			Collection<String> values = valuesMap.values();
+
+			if (values.isEmpty()) {
+				return null;
+			}
 		}
 		else if (returnType.equals(long.class)) {
-			return _typedSettings.getLongValue(method.getName());
+			return _typedSettings.getLongValue(key);
 		}
 		else if (returnType.equals(String.class)) {
-			return _typedSettings.getValue(method.getName());
+			if (getReturnTypeDefault) {
+				return _typedSettings.getValue(key);
+			}
+
+			return _typedSettings.getValue(key, null);
 		}
 		else if (returnType.equals(String[].class)) {
-			return _typedSettings.getValues(method.getName());
+			if (getReturnTypeDefault) {
+				return _typedSettings.getValues(key);
+			}
+
+			return _typedSettings.getValues(key, null);
 		}
 		else if (returnType.isEnum()) {
 			Method valueOfMethod = returnType.getDeclaredMethod(
 				"valueOf", String.class);
 
-			return valueOfMethod.invoke(
-				returnType, _typedSettings.getValue(method.getName()));
+			String value = _typedSettings.getValue(key);
+
+			if (value == null) {
+				return value;
+			}
+
+			return valueOfMethod.invoke(returnType, value);
 		}
 
 		Constructor<?> constructor = returnType.getConstructor(String.class);
 
-		return constructor.newInstance(
-			_typedSettings.getValue(method.getName()));
+		return constructor.newInstance(_typedSettings.getValue(key));
+	}
+
+	private Object _invokeConfigurationOverride(Method method, Object[] args)
+		throws IllegalAccessException, InvocationTargetException,
+			   NoSuchMethodException {
+
+		Class<?> clazz = _configurationOverrideInstance.getClass();
+
+		method = clazz.getMethod(method.getName(), method.getParameterTypes());
+
+		return method.invoke(_configurationOverrideInstance, args);
 	}
 
+	private Object _invokeTypedSettings(Method method)
+		throws NoSuchMethodException, IllegalAccessException,
+			   InvocationTargetException, InstantiationException {
+
+		Class<?> returnType = method.getReturnType();
+
+		Object value = null;
+
+		Meta.AD annotation = method.getAnnotation(Meta.AD.class);
+
+		if ((annotation != null) && !_NULL_VALUE.equals(annotation.id())) {
+			value = _getValue(returnType, annotation.id(), false);
+		}
+
+		if (value == null) {
+			return _getValue(returnType, method.getName(), true);
+		}
+
+		return value;
+	}
+
+	private static final String _NULL_VALUE = "§NULL§";
```

Comment:
Do we really need to hardcode this constant String here? It seems very error prone since the javadoc of annotation.id does not specify that it will return that. So it might change at any moment with a new version of bnd.